src/irr/asset/CGraphicsPipelineLoaderMTL.cpp

```diff
@@ -1,10 +1,393 @@
 #include "irr/asset/CGraphicsPipelineLoaderMTL.h"
 
+namespace
+{
+    constexpr const char* VERT_SHADER_NO_UV_MAIN = 
+R"(
+void main()
+{
+    LocalPos = vPos;
+    gl_Position = irr_glsl_pseudoMul4x4with3x1(CamData.params.MVP, vPos);
+    ViewPos = irr_glsl_pseudoMul3x4with3x1(CamData.params.MV, vPos);
+    mat3 normalMat = irr_glsl_SBasicViewParameters_GetNormalMat(CamData.params);
+    Normal = normalMat*normalize(vNormal);
+}
+)";
+    constexpr const char* FRAG_SHADER_NO_UV_MAIN =
+R"(#define Ia 0.1
+void main()
+{
+    vec3 N = normalize(Normal);
```

Comment:
still need that common builtin to get all the BRDF parameters

Reply:
especially that you've hardcoded the light to be at the camera position (something we want easily overridable)